### ISSUE TITLE
train problem for assignment4

### ISSUE THREAD

**Issue by YanglanWang** [Luvata/CS224N-2019#3]
In assignment4, when executing "sh run.sh train" ,after about 4 or 5 hours, it stops with the information like this:
"
begin validation...
validation: iter 94000, dev . ppl 17.209386
hit patience 5
hit #5 trial
early stop!
"
do you know why?

**Comment by Luvata** [comment]
I think it's not a problem. TLDR : It's early stopping technique when training neural network, to avoid model get overfitting

When training, for every ``valid_niter``, the model will be [evaluated](https://github.com/Luvata/CS224N-2019/blob/9e739b994dfa597a2c17d7a07a4f3bf75d6ad75f/Assignment/a4/run.py#L211) on the validation data , or the dev set (the data that model has never seen before). To make the model generalize well, they check  whether the model performs better on this new data by [compare](https://github.com/Luvata/CS224N-2019/blob/9e739b994dfa597a2c17d7a07a4f3bf75d6ad75f/Assignment/a4/run.py#L216) its validation score with previous validation score. If the model doesn't get better after ``patience`` consecutive validation times, the learning rate will be decayed, as a new trial. After ``max-num-trial`` consecutive decayed times, if the model still doesn't get better on the dev set, we can stop to avoid model getting overfitting
